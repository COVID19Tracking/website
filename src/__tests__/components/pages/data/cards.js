import React from 'react'
import renderer from 'react-test-renderer'
import CasesCard from '~components/pages/data/cards/cases-card'
import HospitalizationCard from '~components/pages/data/cards/hospitalization-card'
import OutcomesCard from '~components/pages/data/cards/outcomes-card'
import AntibodyCard from '~components/pages/data/cards/tests-antibody'
import NationalTestsCard from '~components/pages/data/cards/tests-national'
import ViralTestsCard from '~components/pages/data/cards/tests-viral'

describe('Components : Pages : Data : Cards : Cases', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <CasesCard
          stateSlug="california"
          positive={13}
          positiveIncrease={14}
          sevenDayIncrease={17.3}
          probableCases={4}
          confirmedCases={5}
          national={false}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()

    const nationalTree = renderer
      .create(
        <CasesCard
          positive={13}
          positiveIncrease={14}
          sevenDayIncrease={17.3}
          national
        />,
      )
      .toJSON()
    expect(nationalTree).toMatchSnapshot()
  })
})

describe('Components : Pages : Data : Cards : Hospitalization', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <HospitalizationCard
          stateSlug="california"
          hospitalizedCumulative={4}
          inIcuCumulative={5}
          onVentilatorCumulative={10}
          hospitalizedCurrently={14}
          inIcuCurrently={13}
          onVentilatorCurrently={15}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()

    const nationalTree = renderer
      .create(
        <HospitalizationCard
          stateSlug="california"
          hospitalizedCumulative={4}
          inIcuCumulative={5}
          onVentilatorCumulative={10}
          hospitalizedCurrently={14}
          inIcuCurrently={13}
          onVentilatorCurrently={15}
          national
        />,
      )
      .toJSON()
    expect(nationalTree).toMatchSnapshot()
  })
})

describe('Components : Pages : Data : Cards : Outcomes', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <OutcomesCard
          stateSlug="california"
          deathsLabel="Deaths"
          death={45}
          deathConfirmed={15}
          deathProbable={10}
          recovered={14}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()

    const nationalTree = renderer
      .create(
        <OutcomesCard
          stateSlug="california"
          deathsLabel="Deaths"
          death={45}
          deathConfirmed={15}
          deathProbable={10}
          recovered={14}
          national
        />,
      )
      .toJSON()
    expect(nationalTree).toMatchSnapshot()
  })
})

describe('Components : Pages : Data : Cards : Antibody tests', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<AntibodyCard stateSlug="california" totalTestsAntibody={14} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Components : Pages : Data : Cards : National tests', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <NationalTestsCard
          totalTestResults={14000}
          totalTestResultsIncrease={1400}
          totalTestResulstPercentIncrease={13.4}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Components : Pages : Data : Cards : Viral tests', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ViralTestsCard
          stateSlug="california"
          totalTestEncountersViral={15}
          totalTestsViral={16}
          totalTestsPeopleViral={16}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()

    const treeUnknownUnits = renderer
      .create(
        <ViralTestsCard
          stateSlug="california"
          totalTestEncountersViral={15}
          totalTestsViral={16}
          totalTestsPeopleViral={16}
          unknownUnits
        />,
      )
      .toJSON()
    expect(treeUnknownUnits).toMatchSnapshot()
  })
})
