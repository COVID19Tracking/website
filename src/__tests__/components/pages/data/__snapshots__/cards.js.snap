// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Pages : Data : Cards : Antibody tests renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="header"
  >
    <h3
      className="title"
    >
      Antibody tests
    </h3>
    <span
      className="link"
    >
      <a
        href="/data/state/california/tests-antibody"
      >
        Historical data 
        <span
          className="a11y-only"
        >
           for antibody tests
        </span>
      </a>
    </span>
  </div>
  <div
    className="body"
  >
    <div
      className="statisticWrapper"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Total tests
        </div>
        <div
          className="value"
        >
          14
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Total "
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Components : Pages : Data : Cards : Cases renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="header"
  >
    <h3
      className="title"
    >
      Cases
    </h3>
    <span
      className="link"
    >
      <a
        href="/data/state/california/cases"
      >
        Historical data 
        <span
          className="a11y-only"
        >
           for cases
        </span>
      </a>
    </span>
  </div>
  <div
    className="body"
  >
    <div
      className="statisticWrapper"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Total cases
        </div>
        <div
          className="value"
        >
          13
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Total cases"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
    <div
      className="statisticWrapper subelement"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Confirmed cases
        </div>
        <div
          className="value"
        >
          5
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Confirmed cases"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
    <div
      className="statisticWrapper subelement"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Probable cases
        </div>
        <div
          className="value"
        >
          4
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Probable Cases"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
    <div
      className="drillDown"
    >
      <span
        className="label"
      >
        New cases today
        <div
          aria-hidden={true}
          className="calculated"
        >
           
          (Calculated)
        </div>
      </span>
      <span
        className="value"
      >
        14
      </span>
    </div>
    <div
      className="drillDown"
    >
      <span
        className="label"
      >
        Change over 7 days
        <div
          aria-hidden={true}
          className="calculated"
        >
           
          (Calculated)
        </div>
      </span>
      <span
        className="value"
      >
        1,730
        %+
      </span>
    </div>
  </div>
</div>
`;

exports[`Components : Pages : Data : Cards : Cases renders correctly 2`] = `
<div
  className="card"
>
  <div
    className="header"
  >
    <h3
      className="title"
    >
      Cases
    </h3>
    <span
      className="link"
    >
      <a
        href="/data/national/cases"
      >
        Historical data 
        <span
          className="a11y-only"
        >
           for cases
        </span>
      </a>
    </span>
  </div>
  <div
    className="body"
  >
    <div
      className="statisticWrapper"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Total cases
        </div>
        <div
          className="value"
        >
          13
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Total cases"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
    <div
      className="drillDown"
    >
      <span
        className="label"
      >
        New cases today
        <div
          aria-hidden={true}
          className="calculated"
        >
           
          (Calculated)
        </div>
      </span>
      <span
        className="value"
      >
        14
      </span>
    </div>
    <div
      className="drillDown"
    >
      <span
        className="label"
      >
        Change over 7 days
        <div
          aria-hidden={true}
          className="calculated"
        >
           
          (Calculated)
        </div>
      </span>
      <span
        className="value"
      >
        1,730
        %+
      </span>
    </div>
  </div>
</div>
`;

exports[`Components : Pages : Data : Cards : Hospitalization renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="header"
  >
    <h3
      className="title"
    >
      Hospitalization
    </h3>
    <span
      className="link"
    >
      <a
        href="/data/state/california/hospitalization"
      >
        Historical data
         
        <span
          className="a11y-only"
        >
           for hospitalization
        </span>
      </a>
    </span>
  </div>
  <div
    className="body"
  >
    <div
      className="statisticWrapper"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Ever hospitalized
        </div>
        <div
          className="value"
        >
          4
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Ever hospitalized"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
    <div
      className="statisticWrapper subelement"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Now hospitalized
        </div>
        <div
          className="value"
        >
          14
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Now hospitalized"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
    <div
      className="statisticWrapper"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Ever in ICU
        </div>
        <div
          className="value"
        >
          5
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Ever in ICU"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
    <div
      className="statisticWrapper subelement"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Now in ICU
        </div>
        <div
          className="value"
        >
          13
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Now in ICU"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
    <div
      className="statisticWrapper"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Ever on ventilator
        </div>
        <div
          className="value"
        >
          10
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Ever on ventilator"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
    <div
      className="statisticWrapper subelement"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Now on ventilator
        </div>
        <div
          className="value"
        >
          15
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Now on ventilator"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Components : Pages : Data : Cards : Hospitalization renders correctly 2`] = `
<div
  className="card"
>
  <div
    className="header"
  >
    <h3
      className="title"
    >
      Hospitalization
    </h3>
    <span
      className="link"
    >
      <a
        href="/data/national/hospitalization"
      >
        Historical data
         
        <span
          className="a11y-only"
        >
           for hospitalization
        </span>
      </a>
    </span>
  </div>
  <div
    className="body"
  >
    <div
      className="statisticWrapper"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Currently hospitalized
        </div>
        <div
          className="value"
        >
          14
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Currently hospitalized"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
    <div
      className="statisticWrapper"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Currently in ICU
        </div>
        <div
          className="value"
        >
          13
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Currently in ICU"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
    <div
      className="statisticWrapper"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Currently on ventilator
        </div>
        <div
          className="value"
        >
          15
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Currently on ventilator"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Components : Pages : Data : Cards : National tests renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="header"
  >
    <h3
      className="title"
    >
      Tests
    </h3>
    <span
      className="link"
    >
      <a
        href="/data/national/tests"
      >
        Historical data 
        <span
          className="a11y-only"
        >
           for national tests
        </span>
      </a>
    </span>
  </div>
  <div
    className="body"
  >
    <div
      className="statisticWrapper"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Total tests
        </div>
        <div
          className="value"
        >
          14,000
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Total tests"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
          <div
            className="drillDown"
          >
            <span
              className="label"
            >
              New tests today
              <div
                aria-hidden={true}
                className="calculated"
              >
                 
                (Calculated)
              </div>
            </span>
            <span
              className="value"
            >
              1,400
            </span>
          </div>
          <div
            className="drillDown"
          >
            <span
              className="label"
            >
              Change over 7 days
              <div
                aria-hidden={true}
                className="calculated"
              >
                 
                (Calculated)
              </div>
            </span>
            <span
              className="value"
            >
              1,340
              %+
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Components : Pages : Data : Cards : Outcomes renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="header"
  >
    <h3
      className="title"
    >
      Outcomes
    </h3>
    <span
      className="link"
    >
      <a
        href="/data/state/california/outcomes"
      >
        Historical data 
        <span
          className="a11y-only"
        >
           for outcomes
        </span>
      </a>
    </span>
  </div>
  <div
    className="body"
  >
    <div
      className="statisticWrapper"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Recovered
        </div>
        <div
          className="value"
        >
          14
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Recovered"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
    <div
      className="statisticWrapper"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Deaths
        </div>
        <div
          className="value"
        >
          45
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Deaths"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
    <div
      className="statisticWrapper subelement"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Probable deaths
        </div>
        <div
          className="value"
        >
          10
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Probable deaths"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
    <div
      className="statisticWrapper subelement"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Confirmed deaths
        </div>
        <div
          className="value"
        >
          15
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Confirmed deaths"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Components : Pages : Data : Cards : Outcomes renders correctly 2`] = `
<div
  className="card"
>
  <div
    className="header"
  >
    <h3
      className="title"
    >
      Outcomes
    </h3>
    <span
      className="link"
    >
      <a
        href="/data/national/outcomes"
      >
        Historical data 
        <span
          className="a11y-only"
        >
           for outcomes
        </span>
      </a>
    </span>
  </div>
  <div
    className="body"
  >
    <div
      className="statisticWrapper"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Recovered
        </div>
        <div
          className="value"
        >
          14
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Recovered"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
    <div
      className="statisticWrapper"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Deaths
        </div>
        <div
          className="value"
        >
          45
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Deaths"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
    <div
      className="statisticWrapper subelement"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Probable deaths
        </div>
        <div
          className="value"
        >
          10
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Probable deaths"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
    <div
      className="statisticWrapper subelement"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Confirmed deaths
        </div>
        <div
          className="value"
        >
          15
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Confirmed deaths"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Components : Pages : Data : Cards : Viral tests renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="header"
  >
    <h3
      className="title"
    >
      Viral (PCR) tests
    </h3>
    <span
      className="link"
    >
      <a
        href="/data/state/california/tests-viral"
      >
        Historical data
         
        <span
          className="a11y-only"
        >
           for viral (PCR) tests
        </span>
      </a>
    </span>
  </div>
  <div
    className="body"
  >
    <div
      className="statisticWrapper"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Total tests
          <br />
          (in test encounters)
        </div>
        <div
          className="value"
        >
          15
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Total tests (test encounters)"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
    <div
      className="statisticWrapper"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Total tests
          <br />
          (in specimens)
        </div>
        <div
          className="value"
        >
          16
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Total tests (specimens)"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
    <div
      className="statisticWrapper"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Total tests
          <br />
          (in people)
        </div>
        <div
          className="value"
        >
          16
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Total tests (people)"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Components : Pages : Data : Cards : Viral tests renders correctly 2`] = `
<div
  className="card"
>
  <div
    className="header"
  >
    <h3
      className="title"
    >
      Viral (PCR) tests
    </h3>
    <span
      className="link"
    >
      <a
        href="/data/state/california/tests-viral"
      >
        Historical data
         
        <span
          className="a11y-only"
        >
           for viral (PCR) tests
        </span>
      </a>
    </span>
  </div>
  <div
    className="body"
  >
    <div
      className="statisticWrapper"
    >
      <div
        className="statistic"
      >
        <div
          className="title"
        >
          Total tests
          <br />
          (in unclear units)
        </div>
        <div
          className="value"
        >
          16
        </div>
        <div
          className="info"
        >
          <button
            aria-label="Definition of Total tests (unclear units)"
            className="link"
            onClick={[Function]}
            type="button"
          >
            Definition
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
